library/src/interactor_impl.cxx

```diff
@@ -21,16 +21,45 @@
 #include <vtkStringArray.h>
 #include <vtkVersion.h>
 #include <vtksys/SystemTools.hxx>
+#include <vtkPicker.h>
+#include <vtkTransform.h>
 
 #include <chrono>
 #include <cmath>
 #include <map>
 
+#include "camera.h"
+
 namespace f3d::detail
 {
-class interactor_impl::internals
-{
-public:
+  //----------------------------------------------------------------------------
+  // Method defined to normalize the Z axis so all models are treated temporarily
+  // as Z-up axis models. 
+  vtkNew<vtkTransform> zUpTransform(vtkRenderer* renderer)
+  {
+    const double* up = renderer->GetEnvironmentUp();
+    const double* right = renderer->GetEnvironmentRight();
+    double fwd[3];
+    vtkMath::Cross(right, up, fwd);
+
+    const double m[16] = {
+      right[0], right[1], right[2], 0, //
+      fwd[0], fwd[1], fwd[2], 0,       //
+      up[0], up[1], up[2], 0,          //
+      0, 0, 0, 1,                      //
+    };
+    vtkNew<vtkTransform> toZup;
+    toZup->SetMatrix(m);
+    return toZup;
+  }
+  vtkNew<vtkTransform> zUpTransform(vtkRenderWindowInteractor* interactor)
```

Comment:
these two methods API is not expected. A single method like this would do the trick:

`void zUpTransform(vtkRenderer* renderer, vtkTransform* transform)`

Then use it like that:

```
vtkNew<vtkTransform> transform;
zUpTransform(randerer, transform);
```

Reply:
and make it a static method of interactor_impl::internals for simplicity, I think.